### PR TITLE
Added Alpine dev container for Go development

### PR DESCRIPTION
**Pull request by qdm12**
- Uses VS code remote container development extension
- All Go and dependencies are in a Docker container
- Quick startup once the container image is built
- Recommended extensions added

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        "shardulm94.trailing-spaces"
+    ],
+    "settings": {
+        "go.gopath": "/go",
+        "go.inferGopath": false,
+        "go.useLanguageServer": true
+    },
+    "postCreateCommand": "go mod download",
+    "runArgs": [
+        // Comment out the next line to run as root instead.
```
Comment (mjarkk) [inline .devcontainer/devcontainer.json]: The github file viewer doesn't like this line,  
maybe just remove it

Comment (mjarkk) [reply .devcontainer/devcontainer.json]: Oke then we will leaf it here

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        "ms-vscode.go",
+        "davidanson.vscode-markdownlint",
+        "shardulm94.trailing-spaces"
+    ],
+    "settings": {
+        "go.gopath": "/go",
+        "go.inferGopath": false,
+        "go.useLanguageServer": true
+    },
+    "postCreateCommand": "go mod download",
```
[the remark this answers is not in the document]
Comment (dawidd6) [reply .devcontainer/devcontainer.json]: I don't use vscode for Go dev, but I know @jesseduffield does, so maybe he can say something about it.

[the remark this answers is not in the document]
Comment (jesseduffield) [reply .devcontainer/devcontainer.json]: @qdm12 my VS code finds the folder fine

image

Perhaps you have a global `.ignore` or `.gitignore` file that ignore `vendor/` directories? There is a setting that will check for those:

image

Likewise you might have the vendor directory added to File Exclude or Search Exclude:

image

**Comment by jesseduffield** [comment]
closing due to staleness but let me know if you want to pick this up again @qdm12

**Pull request closed without merging**